### DISCUSSION TITLE
Error when deleting required objects/arrays

### DISCUSSION THREAD

**Discussion by laurajinks** [CarlosNZ/json-edit-react#142] [Q&A]
Hi, I'm trying to implement an editor with validation using ajv as outlined in the demos. I have it working properly when validating editing or deleting individual required values, but I'm getting an error when deleting required objects or arrays. It seems to stem from this line here https://github.com/CarlosNZ/json-edit-react/blob/main/src/CollectionNode.tsx#L196

For example, if I had a JSON object like:

```
{
  "foo": {
    "bar": "baz"
  }
}
```

and `bar` was a required field, I am seeing the error messaging correctly. However if `foo` is a required field and I attempt to delete it, I am getting this error:

```
Error: Unable to extract object property
Looking for property: foo
```

Additionally, this is less of a big deal, but if I edit the key `foo` and it fails validation, It correctly does not persist the edit, however it does not show an inline error message like it does when incorrectly editing a string value.

Any thoughts on something I could be missing here?

**Comment by CarlosNZ** [comment]
Thanks for pointing these out, it seems like a bit of a bug. I'll look into it very soon. Thanks for providing very specific examples.

**Comment by CarlosNZ** [comment]
Moved to issue #143

**Comment by CarlosNZ** [comment]
Hi @laurajinks, this should be fixed now in v1.19.1, please let me know if it works for you now.

> Additionally, this is less of a big deal, but if I edit the key `foo` and it fails validation, It correctly does not persist the edit, however it does not show an inline error message like it does when incorrectly editing a string value.

I think this is also fixed now -- unless I'm missing what you're saying?

**Comment by CarlosNZ** [comment]
Seems fixed to me
